src/librustc_codegen_llvm/back/write.rs

```diff
@@ -2586,8 +2586,8 @@ fn create_msvc_imps(cgcx: &CodegenContext, llcx: &llvm::Context, llmod: &llvm::M
                 imp_name.extend(name.to_bytes());
                 let imp_name = CString::new(imp_name).unwrap();
                 (imp_name, val)
-            })
-            .collect::<Vec<_>>();
+            });
+
```

Comment:
You're iterating over globals here and then adding new globals in the loop below. With the collect that's fine, as you'll only iterate existing globals. Without the collect this is going to be an infinite loop and you OOM.

Reply:
Weird that this doesn't seem to cause issues on architectures different than `i686` (at least that's the one the OOM was being hit on appveyor); I can compile with these changes on `x64` without issues.